### ISSUE TITLE
Decode perf: skip logsumexp in greedy argmax (fast path)

### ISSUE THREAD

**Issue by dahai80** [Blaizzy/mlx-vlm#1532]
## Motivation

During greedy decoding, the model forward produces logits and the sampling path normalizes them with `logsumexp` (per-row, in `log_softmax`). For **greedy / argmax** sampling the result is invariant to this constant per-row offset:

```
argmax(logits) == argmax(logits - logsumexp(logits))
```

so the `logsumexp` reduction is wasted work on the greedy decode path — which is the common case for VLM image-captioning / VQA loops.

## Proposal

Add a fast path in `mlx_vlm.models.base` / `ar.py` greedy sampling that skips the `logsumexp` reduction, guarded by the sampling mode (greedy only). Implementation in #1512.

## Correctness

The argmax invariant above guarantees identical token selection. Numerically, `logsumexp` is a per-row scalar subtracted from every element; `argmax` is invariant to any constant offset, so greedy output is bit-identical.

## Performance (measured)

Benefit is **overlap-dependent**. We measured the analogous fast path in mlx-lm:

- **Batched decode path: ~0–5% wall-clock.** `mx.async_eval` already overlaps the `logsumexp` reduction with the next step's forward pass, so removing it from the critical graph yields little on the batched path that #1512 implements.
- **Non-batched (sequential) path: ~5–8%.** See ml-explore/mlx-lm#1479 for the non-batched measurement.

The VLM decode loop uses the same `async_eval` structure as mlx-lm's batched path, so we expect similar overlap-dependent behavior (~0–5%). The main win is removing the `logsumexp` allocation/reduction from the graph and avoiding the wait when overlap is not possible (small batch / no async_eval). Happy to run VLM-specific benchmarks on request.

## Context

This is item #1 of the omnibus decode-perf issue #1482 (closed for now). Opening this focused issue per maintainer request in #1512 ("please open a issue first") so the PR has a linked issue to track.

Refs #1482.
